pkg/objstore/client/factory.go

```diff
@@ -81,5 +85,19 @@ func NewBucket(logger log.Logger, confContentYaml []byte, reg prometheus.Registe
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("create %s client", bucketConf.Type))
 	}
-	return objstore.NewTracingBucket(objstore.BucketWithMetrics(bucket.Name(), bucket, reg)), nil
+
+	var prefixedBucket objstore.Bucket
+	if validPrefix(bucketConf.Config.Prefix) {
```

Comment:
Although this seems a micro perf improvement, I guess it has the advantage of reducing risks of bugs for those who don't want to use prefixes (while also increasing perf for those).

Reply:
If prefix is configured but it is invalid, should we catch that error case and print the error log?

Reply:
I liked your idea :)
But I don't have an idea of how I would differentiate between a not filled key and a non-existent key. would love to have more ideas on this one.